cmd/istioctl/main.go

```diff
@@ -41,9 +41,8 @@ const (
 var (
 	platform string
 
-	kubeconfig     string
-	namespace      string
-	istioNamespace string
```

Comment:
so it's not needed anywhere (like kube-inject) ? how will it find the istio components ? (/what creates the meshconfig for a given namespace ?)

Reply:
You have a point. As a previous step, I was creating the config map in every namespace with kubectl, but maybe we can do it programatically, in which case I need to revert this change.

Reply:
I would think it'd be best to not denormalize the config map but have pointers in each managed namespace to which namespace is the istio one (managing the namespace)
ie only have 1 required istio-namespace property ? technically you don't even need to remember after injection or do you need that map at runtime in sidecars somehow ?

Reply:
Thanks for bringing this up (the first part), it could affect the ux.
I added code in istioctl kube-inject to copy the configMap in the app namespace, until I have time to properly implemented with a namespace controller:
https://github.com/istio/pilot/issues/1153

We cannot de-normalize, it would still mean cross-namespace, and we need to map the actual data.